src/api/category/services/getMany.js

```diff
@@ -7,13 +7,23 @@ async function readMany(req, res) {
   try {
     const limit = +req.query.limit;
     const pageNo = +req.query.pageNo;
+    const { language } = req.headers;
+    const { search } = req.query;
+    const { notSelectedLanguages } = req;
     const filter = { deleted: false };
+    const dispatchedLanguage = { name: `name${req.headers.language}` };
+
+    const select = `-name${notSelectedLanguages[0]} -name${notSelectedLanguages[1]}`;
+    filter['$or'] = [
+      { [`name${language}`]: { $regex: `${search}`, $options: 'i' } },
+      { [`name${notSelectedLanguages[0]}`]: { $regex: `${search}`, $options: 'i' } },
+      { [`name${notSelectedLanguages[1]}`]: { $regex: `${search}`, $options: 'i' } },
+    ];
```

Comment:
You are using { $regex: `${search}`, $options: 'i' } object many times. Isn't it easier to define a variable with this value, ex: const regexpSearch = { $regex: `${search}`, $options: 'i' }; and us it everywhere?